### PR TITLE
Removing .setInner() from public access; changing CreatedResources into Map; sample enhancements

### PR DESCRIPTION
**Pull request by martinsawicki**
- users should not be able to call .setInner(), and we already documented it as "internal use only". It is not needed on the interface
- `CreatedResources` is now a map for better discoverability of how to get a created resource based on the key of the creatable definition
- updated (and enhanced) the samples affected by this

Flagging as compat-breaks relative to beta4.1 since:
- we're removing the `.setInner()` method, but in reality users should not have written code against it because it was documented as "internal use only"
- `CreatedResources` returned by the parallel `create()` is now a `Map<String, T>` indexed by the key of the definition ("creatable")

### CODE REVIEW

File: azure-samples/src/main/java/com/microsoft/azure/management/trafficmanager/samples/ManageSimpleTrafficManager.java [hunk 1]
```diff
@@ -117,10 +117,12 @@ public static void main(String[] args) {
                         .withPerformanceBasedRouting();
 
                 TrafficManagerProfile.DefinitionStages.WithCreate profileWithCreate = null;
+                int i = 1;
```
Comment (anudeepsharma) [inline azure-samples/src/main/java/com/microsoft/azure/management/trafficmanager/samples/ManageSimpleTrafficManager.java]: >i [](start = 20, length = 1)

name it priorityIndex

File: azure-mgmt-redis/src/main/java/com/microsoft/azure/management/redis/implementation/RedisCacheImpl.java [hunk 2]
```diff
@@ -479,7 +479,7 @@ public void call(RedisCache redisCache) {
                                 break;
                             }
                             innerResource = client.get(resourceGroupName(), name());
-                            redisCache.setInner(innerResource);
+                            ((RedisCacheImpl) redisCache).setInner(innerResource);
```
Comment (anudeepsharma) [inline azure-mgmt-redis/src/main/java/com/microsoft/azure/management/redis/implementation/RedisCacheImpl.java]: IMO innerToFluentMap can be used and we can avoid the casting

**Comment by anudeepsharma** [comment]
Looks good to me with 2 minor comments.

File: azure-samples/src/main/java/com/microsoft/azure/management/trafficmanager/samples/ManageSimpleTrafficManager.java (hunk 1 above)
Comment (martinsawicki) [inline azure-samples/src/main/java/com/microsoft/azure/management/trafficmanager/samples/ManageSimpleTrafficManager.java]: ok, done

File: azure-mgmt-redis/src/main/java/com/microsoft/azure/management/redis/implementation/RedisCacheImpl.java (hunk 2 above)
Comment (martinsawicki) [inline azure-mgmt-redis/src/main/java/com/microsoft/azure/management/redis/implementation/RedisCacheImpl.java]: as discussed, innerToFluentMap won't work

**Comment by martinsawicki** [comment]
@anuchandy, since i wound up piling more changes onto the previous PR (due to issues with our CI), it'd be great if you could also take a look - in particular the CreatedResources changes

**Pull request merged**